### ISSUE TITLE
Refactor Hyperparameter classes

### ISSUE THREAD

**Issue by zuoxingdong** [zuoxingdong/lagom#207]
Inspired by Amazon SageMaker and Ray, try to refactor the classes for hyperparameters. 

e.g. 
- Categorical
- Continuous
- Logarithmic: for small scale, e.g. learning rate
